From b3bb216229146b6dcc522b9b7e2d1f2435cacf71 Mon Sep 17 00:00:00 2001
From: Christoph Reiter <reiter.christoph@gmail.com>
Date: Thu, 26 Aug 2021 05:14:18 +0200
Subject: [PATCH] build: fix implicit-fallthrough errors with clang

The clang build fails due to -Werror=implicit-fallthrough being
on by default and some fallthrough cases not being marked as such.

Use G_GNUC_FALLTHROUGH or duplicate the code in those cases.
---
 gtk/gtkimcontextime.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/gtk/gtkimcontextime.c b/gtk/gtkimcontextime.c
index c8378c836e..102eccec38 100644
--- a/gtk/gtkimcontextime.c
+++ b/gtk/gtkimcontextime.c
@@ -667,6 +667,7 @@ gtk_im_context_ime_focus_in (GtkIMContext *context)
   switch (context_ime->priv->focus_behavior)
     {
       case GTK_WIN32_IME_FOCUS_BEHAVIOR_COMMIT:
+        G_GNUC_FALLTHROUGH;
       case GTK_WIN32_IME_FOCUS_BEHAVIOR_DISCARD:
         gtk_im_context_ime_reset (context);
         break;
@@ -724,7 +725,7 @@ gtk_im_context_ime_focus_out (GtkIMContext *context)
             context_ime->priv->pretend_empty_preedit = FALSE;
             g_free (utf8str);
           }
-        /* fallthrough */
+        G_GNUC_FALLTHROUGH;
       case GTK_WIN32_IME_FOCUS_BEHAVIOR_DISCARD:
         gtk_im_context_ime_reset (context);
 
@@ -1054,7 +1055,7 @@ gtk_im_context_ime_message_filter (GdkWin32Display *display,
         default:
           break;
         }
-
+      break;
     default:
       break;
     }
-- 
GitLab

